packages/flutter_tools/lib/src/ios/ios_workflow.dart

```diff
@@ -125,66 +128,74 @@ class IOSValidator extends DoctorValidator {
       }
     }
 
-    // brew installed
-    if (hasHomebrew) {
-      brewStatus = ValidationType.installed;
+    int checksFailed = 0;
 
-      if (!iMobileDevice.isInstalled) {
-        brewStatus = ValidationType.partial;
-        messages.add(ValidationMessage.error(
-            'libimobiledevice and ideviceinstaller are not installed. To install, run:\n'
-            '  brew install --HEAD libimobiledevice\n'
-            '  brew install ideviceinstaller'
-        ));
-      } else if (!await iMobileDevice.isWorking) {
-        brewStatus = ValidationType.partial;
+    if (!iMobileDevice.isInstalled) {
+      checksFailed += 1;
+      packageManagerStatus = ValidationType.partial;
+      messages.add(ValidationMessage.error(
+        'libimobiledevice and ideviceinstaller are not installed. To install with Brew, run:\n'
+        '  brew install --HEAD libimobiledevice\n'
+        '  brew install ideviceinstaller'
+      ));
+    } else if (!await iMobileDevice.isWorking) {
+      checksFailed += 2;
```

Comment:
+= 1?

Reply:
No, because it's an else if, it means that the above check already failed, so we want +2 and +3 respectively.

Reply:
The check above has to pass (libimobile device is installed) - should both of these branches be considered part of the same "check"?

Reply:
I mean, I suppose we can separate them into their own if statements? Is that what you're saying?